README.rst

```diff
@@ -42,16 +63,137 @@ Add the AutoExtract downloader middleware in the settings file::
+            # We're making two requests here:
+            # - one through Scrapy to build the response argument
+            # - another through providers to build the article argument
+            yield article.to_item()
+
+Note that on the example above, we're going to perform two requests:
+
+* one goes through Scrapy (it might use Crawlera, Splash or no proxy at all, depending on your configuration)
+* another goes through AutoExtract API using `scrapinghub-autoextract`_
+
+If you don't need the additional request going through Scrapy,
+you can annotate the response argument of your callback with ``DummyResponse``.
+This will ignore the Scrapy request and only the AutoExtract API will be fetched.
+
+For example::
+
+    import scrapy
+    from autoextract_poet.page_inputs import AutoExtractArticleData
+    from scrapy_poet.utils import DummyResponse
+
+    class SampleSpider(scrapy.Spider):
+
+        name = "sample"
+
+        def parse(self, response: DummyResponse, article: AutoExtractArticleData):
+            # We're making a single request here to build the article argument
+            yield article.to_item()
+
+Configuration
+^^^^^^^^^^^^^
+
+First, you need to configure scrapy-poet as described on `scrapy-poet's documentation`_
+and then enable AutoExtract providers by putting the following code to Scrapy's ``settings.py`` file::
+
+    # Install AutoExtract providers
+    import scrapy_autoextract.providers
+    scrapy_autoextract.providers.install()
+
+    # Enable scrapy-poet's provider injection middleware
+    DOWNLOADER_MIDDLEWARES = {
+        'scrapy_poet.InjectionMiddleware': 543,
+    }
+
+    # Configure Twisted's reactor for asyncio support on Scrapy
+    TWISTED_REACTOR = 'twisted.internet.asyncioreactor.AsyncioSelectorReactor'
+
+Currently, our providers are implemented using asyncio.
+Scrapy has introduced asyncio support since version 2.0
+but as of Scrapy 2.3 you need to manually enable it by configuring Twisted's default reactor.
+Check `Scrapy's asyncio documentation`_ for more information.
+
+Checklist:
+
+* scrapy-poet is installed and downloader/injector middleware is configured
+* autoextract-poet is installed (page inputs are imported from this lib)
+* providers are installed on settings.py
+* Scrapy's asyncio support is enabled on settings.py
+
+Now you should be ready to use our AutoExtract providers.
+
+Exceptions
+^^^^^^^^^^
+
+While trying to fetch AutoExtract API, providers might raise some exceptions.
+Those exceptions might come from scrapy-autoextract providers themselves,
+`scrapinghub-autoextract`_, or Tenacity, the library used to implement retries.
+For example:
+
+* ``scrapy_poet.providers.QueryError``: raised when a `Query-level error`_ is returned even after retrying the request
+* ``autoextract.aio.errors.RequestError``: raised when a `Request-level error`_ is returned
+* ``autoextract.aio.errors.QueryRetryError``: raised when it's not possible to retry a `Query-level error`_
```

Comment:
I think this one is never raised. Should be removed from here.

Reply:
I've removed the line that cites QueryError exceptions because they never reach userland's code. 
QueryRetryError does.

More details on the related commit's message.

Reply:
@victor-torres I think that `QueryRetryError` also never reaches the user. See this lines of code: https://github.com/scrapinghub/scrapinghub-autoextract/blob/master/autoextract/aio/client.py#L259-L263

The exception is captured, and just the last result is returned without reraising the exception. 

But maybe I'm missing something.

Reply:
This seems consistent with the `request_raw` documentation: https://github.com/scrapinghub/scrapinghub-autoextract/blob/master/autoextract/aio/client.py#L157-L163

Reply:
That's very true. We've discussed this during our call the other day but I totally forgot about it. Thanks for pointing it. Fixed/removed.